### PR TITLE
TTL API for logs V2

### PR DESCRIPTION
**Pull request by nityanandagohain**
TTL API for logs V2

Fixes https://github.com/SigNoz/signoz/issues/5783

### CODE REVIEW

**Comment by srikanthccv** [comment]
Let me know when this is ready and resolve conflicts

**Comment by nityanandagohain** [comment]
this is ready for review.

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 1]
```diff
@@ -1996,6 +1996,107 @@ func getLocalTableName(tableName string) string {
...
+	ttlPayload := map[string]string{
+		tableNameArray[0]: ttlLogsV2,
+		tableNameArray[1]: ttlLogsV2Resource,
+	}
+
+	// set the ttl if nothing is pending/ no errors
+	go func(ttlPayload map[string]string) {
+		for tableName, query := range ttlPayload {
+			// we don't want the ttl query to timeout
+			query += " SETTINGS distributed_ddl_task_timeout = -1"
```
Comment (srikanthccv) [inline pkg/query-service/app/clickhouseReader/reader.go:2054]: Please add `materialize_ttl_after_modify=0` instead as mentioned in https://github.com/SigNoz/signoz/issues/5470. We will update other tables in separate PRs.

Comment (nityanandagohain) [reply pkg/query-service/app/clickhouseReader/reader.go]: thanks have made the change

**Review by srikanthccv — APPROVED** [review]

**Comment by prashant-shahi** [comment]
@srikanthccv We will need to keep the use of `materialize_ttl_after_modify` consistent across tables of all signals (logs, traces, and metrics).

Followed by documenting this behaviour in our website retention page.

**Pull request merged**